editoast/src/generated_data/sql/generate_signal_layer.sql

```diff
@@ -32,18 +32,24 @@ INSERT INTO infra_layer_signal (
     )
 SELECT collect.signal_id,
     $1,
-    degrees(
-        ST_Azimuth(
-            ST_LineInterpolatePoint(track_geo, GREATEST(norm_pos - 0.0001, 0.)),
-            ST_LineInterpolatePoint(track_geo, LEAST(norm_pos + 0.0001, 1.))
-        )
-    ) + angle_direction,
-    degrees(
-        ST_Azimuth(
-            ST_LineInterpolatePoint(track_sch, GREATEST(norm_pos - 0.0001, 0.)),
-            ST_LineInterpolatePoint(track_sch, LEAST(norm_pos + 0.0001, 1.))
-        )
-    ) + angle_direction,
+    COALESCE(
+        degrees(
+            ST_Azimuth(
+                ST_LineInterpolatePoint(track_geo, GREATEST(norm_pos - 0.0001, 0.)),
```

Comment:
Probably too complex for the gain, but an alternate idea could be to get previous/next (non-point) geometry.

Reply:
To know the next point, we need the topological graph. Moreover, it is not guaranteed that there will be a next/previous point. For these reasons, it seems simpler to leave the current behaviour.